### PR TITLE
Add OIDs for Edwards curves

### PR DESCRIPTION
**Pull request by MatthiasValvekens**
This PR adds support for the OIDs defined in [RFC 8410](https://tools.ietf.org/html/rfc8410). I'm mainly interested in the EdDSA use case, but I also went ahead and added the OIDs for Edwards curve ECDH keys.

Let me know if there's anything I need to fix before this commit can be merged!

Edit: not sure why the py26 check is failing --- the test suite ran fine on my machine, and I essentially only added keys to dictionaries; I'm not seeing how that could cause issues with older Python versions... I can't view the logs, but could that be caused by the infamous TLS 1.0/1.1 deprecation issue with PyPI for Python <2.7?

### CODE REVIEW

**Comment by joernheissler** [comment]
Hi, could you please add test cases, e.g. private keys and self-signed certs that were generated with openssl?

**Comment by joernheissler** [comment]
> not sure why the py26 check is failing --- the test suite ran fine on my machine, and I essentially only added keys to dictionaries; I'm not seeing how that could cause issues with older Python versions... I can't view the logs, but could that be caused by the infamous TLS 1.0/1.1 deprecation issue with PyPI for Python <2.7?

Can't see the logs either. It appears the circleci would like write access to all my stuff. Not going to happen.
The tests also include other packages which try to access public servers for TLS testing. So it's perfectly possible that something is incompatible.

**Comment by MatthiasValvekens** [comment]
> Hi, could you please add test cases, e.g. private keys and self-signed certs that were generated with openssl?

D'oh, of course. I'll take care of that. I actually have test cases that I generated for use with the code in my own project that monkeypatches these values into `asn1crypto`, but I forgot to add them to the PR.

**Comment by MatthiasValvekens** [comment]
Hi @joernheissler, one quick question: the public / private key fields in a PublicKeyInfo  / PrivateKeyInfo for an EdDSA key should be treated as opaque bit strings / octet strings (respectively). I took this as meaning that these fields would encapsulate a tagged OctetString value, but apparently that was wrong: the key is encoded without any tags, and in particular the contents of this field aren't valid ASN.1 most of the time. See [here](https://lapo.it/asn1js/#MIIBLDCB36ADAgECAghWAUdKKo3DMDAFBgMrZXAwGTEXMBUGA1UEAwwOSUVURiBUZXN0IERlbW8wHhcNMTYwODAxMTIxOTI0WhcNNDAxMjMxMjM1OTU5WjAZMRcwFQYDVQQDDA5JRVRGIFRlc3QgRGVtbzAqMAUGAytlbgMhAIUg8AmJMKdUdIt93LQ-91oNvzoNJjga9OukqY6qm05qo0UwQzAPBgNVHRMBAf8EBTADAQEAMA4GA1UdDwEBAAQEAwIDCDAgBgNVHQ4BAQAEFgQUmx9e7e0EM4Xk97xiPFl1uQvIuzswBQYDK2VwA0EAryMB_t3J5v_BzKc9dNZIpDmAgs3babFOTQbs-BolzlDUwsPrdGxO3YNGhW7Ibz3OGhhlxXrCe1Cgw1AH9efZBw) for an example from RFC 8410.


Is there a way to make `ParsableOctetBitString` and `ParsableOctetString` in `asn1crypto` deal with these things gracefully? Otherwise `.native` will pretty much always break on these certificates.


Since I just pass through public key info to the library that handles signing in my project, I didn't notice this issue until now --- thanks for making me check ;)

**Comment by MatthiasValvekens** [comment]
I've amended the PR with some tests now. The key byte comparisons were based on output from OpenSSL.

Thank you for the review, and sorry for not catching that encoding issue with the public keys earlier!

**Review by joernheissler — APPROVED** [review]

**Comment by joernheissler** [comment]
Circleci complains `failed to create host: Image xcode:9.0.1 is not supported`.

Your PR looks good, thanks!

File: asn1crypto/algos.py [hunk 1]
```diff
@@ -371,6 +378,8 @@ def hash_algo(self):
             'sha256_ecdsa': 'sha256',
             'sha384_ecdsa': 'sha384',
             'sha512_ecdsa': 'sha512',
+            'ed25519': 'sha512',
+            'ed448': 'shake256',
```
Comment (saper) [inline asn1crypto/algos.py:382]: Late to the party - I wonder if those values are correct. Those hashes are used deeply internally and should not be passed to as a parameter to signature generation/verification functions.

I have proposed setting "raw" value here in https://github.com/wbond/asn1crypto/pull/230 - up for discussion!

**Pull request merged**